File: script/models/cap_subject.py
from enum import Enum


class CAPSubject(Enum):
    CHINESE = "Chinese"
    ENGLISH_Reading = "English"
    ENGLISH_LISTENING = "Listening"
    MATH = "Math"
    NATURE = "Nature"
    SOCIETY = "Society"

    def get_chinese_name(self) -> str:
        match self:
            case self.CHINESE:
                return "國文科"
            case self.ENGLISH_Reading:
                return "英文科 (閱讀)"
            case self.ENGLISH_LISTENING:
                return "英文科 (聽力)"
            case self.MATH:
                return "數學科"
            case self.NATURE:
                return "自然科"
            case self.SOCIETY:
                return "社會科"

    def get_duration(self) -> int:
        match self:
            case self.CHINESE:
                return 70
            case self.ENGLISH_Reading:
                return 60
            case self.ENGLISH_LISTENING:
                return 25
            case self.MATH:
                return 80
            case self.NATURE:
                return 70
            case self.SOCIETY:
                return 70

File: lib/src/exam/cap_subject.dart
enum CAPSubject {
  chinese('Chinese', '國文科'),
  englishReading('English', '英文科（閱讀）'),
  englishListening('Listening', '英文科（聽力）'),
  math('Math', '數學科'),
  nature('Nature', '自然科'),
  society('Society', '社會科');

  final String id;
  final String chineseName;

  const CAPSubject(this.id, this.chineseName);
}
